CPAC/nuisance/utils/compcor.py

```diff
@@ -181,9 +181,34 @@ def fallback_svd(a, full_matrices=True, compute_uv=True):
 
 
 def TR_string_to_float(tr):
-    if 'ms' in tr:
-        tr = float(tr.replace('ms', '')) / 1000
-    else:
-        tr = float(tr.replace('s', ''))
+    """
+    Convert TR string to seconds (float). Suffixes 's' or 'ms' to indicate
+    seconds or milliseconds.
+
+    Parameters
+    ----------
+    tr : TR string representation. May use suffixes 's' or 'ms' to indicate
+    seconds or milliseconds.
+
+    Returns
+    -------
+    tr in seconds (float)
+    """
+    if not isinstance(tr, str):
+        raise TypeError(f'Improper type for TR_string_to_float ({tr}).')
+
+    tr_str = tr.replace(' ', '')
+    factor = 1.
+
+    if tr_str.endswith('ms'):
+        tr_str = tr_str[:-2]
+        factor = 0.001
+    elif tr.endswith('s'):
+        tr_str = tr_str[:-1]
+
+    try:
+        tr_numeric = float(tr_str)
+    except Exception as exc:
```

Comment:
A broad exception is definitely better than no error handling, but what are we expecting to fail here if we've already checked type (or `is None`)? Especially since this is downstream of https://github.com/FCP-INDI/C-PAC/blob/44e2431785cdb953fd8aa53c8767f53c1dfeb810/CPAC/utils/utils.py#L974-L977 https://github.com/FCP-INDI/C-PAC/blob/44e2431785cdb953fd8aa53c8767f53c1dfeb810/CPAC/utils/utils.py#L983

Reply:
If we trust the inputs we should document the intended usage and maybe indicate that it's a module private function.